src/rest-utils.h

```diff
@@ -20,8 +20,26 @@
 #ifndef REST_UTILS_H
 #define REST_UTILS_H
 
+#include <stdlib.h>
+#include <stdint.h>
+
+typedef struct device_database_t
```

Comment:
Implementing new datatypes increases code complexity and issues probability.

In my opinion, `rest_list_t` might be used, where you could insert `device_database_entry_t` could be used (therefore it might eliminate some new functions)

My insights should be reviewed by other project members, may be I am wrong.

Reply:
This could be done if 'device_database_t' used static memory for storing it's entries. And that is something worth to discuss. But if we leave dynamic memory, we still need functions to manage it all, and only 'alloc_device_list' could be superseded by 'rest_list_new' and 'rest_list_add' in a loop. Furthermore, a 'rest_list_t' holding a list of 'device_database_t' couldn't be used with 'rest_list_remove' and 'rest_list_delete', since this would cause memory leaks. I see unnecessary overhead in this solution. Any thoughts @GiedriusM ?

Reply:
> Furthermore, a 'rest_list_t' holding a list of 'device_database_t' couldn't be used with 'rest_list_remove' and 'rest_list_delete', since this would cause memory leaks.

Perhaps I don't understand your design ideas properly, but correct me if I'm wrong:

Example fragment 1
```
// DB init
rest_list_t *database = rest_list_new(); // Creating new database
database_entry_t *db_entry = malloc(sizeof(database_entry_t)); // Allocate db entry
db_entry->property = malloc(sizeof(db_entry_property_t)); // Allocate db entry property
rest_list_add(database, db_entry); // Add entry to database
```

Example fragment 2
```
// DB entry freeing function
free(db_entry->property);
```

Example fragment 3
```
// DB deinit
rest_list_entry_t *entry;
database_entry_t *db_entry;
for (entry = list->head; entry != NULL; entry = entry->next)
{
    free_db_entry((db_entry_t *) entry->data);
}
rest_list_delete(database);
```

Provided example has no multithreading problems, neighter I see leaks.
@taubel, I'm starting a discussion for a reason, just saw you use linked-list, Giedrius created one with mutex locks a while ago, and did that for a reason (multithreading caused problems).

Reply:
What I meant is that individual mallocing and freeing is still needed, as you showed in your example, which means that neither we decrease the number of functions, neither the number of datatypes. I will contemplate on the need of mutexes, but I guess it would make sense to add them

Reply:
@taubel, alright, just tried to give some insight, perhaps you notice, that you will have specify mutex locking, unlocking, etc. during manipulation of data inside your functions, which will be copy of existing functions (for example `rest_list_add` ).